core/src/unprocessed_packet_batches.rs

```diff
@@ -79,6 +90,50 @@ impl UnprocessedPacketBatches {
     pub fn with_capacity(capacity: usize) -> Self {
         UnprocessedPacketBatches(VecDeque::with_capacity(capacity))
     }
+
+    /// Returns total number of all packets (including unprocessed and processed) in buffer
+    #[allow(dead_code)]
+    fn get_packets_count(&self) -> usize {
+        self.iter()
+            .map(|deserialized_packet_batch| deserialized_packet_batch.packet_batch.packets.len())
+            .sum()
+    }
+
+    /// Returns total number of unprocessed packets in buffer
+    #[allow(dead_code)]
+    fn get_unprocessed_packets_count(&self) -> usize {
+        self.iter()
+            .map(|deserialized_packet_batch| deserialized_packet_batch.unprocessed_packets.len())
+            .sum()
+    }
+
+    /// Iterates packets in buffered packet_batches, returns all unprocessed packet's stake,
```

Comment:
nit: maybe define what an unprocessed packet's stake means? Something like:

`/// Iterates the inner `Vec<DeserializedPacketBatch>`. Returns the flattened result of mapping each `DeserializedPacketBatch` to a list the batch's inner packets' sender's stake and their `PacketLocator`'s within the `Vec<DeserializedPacketBatch>`.

Reply:
nice, copy-pasted in fullness 😜